mne/io/brainvision/brainvision.py

```diff
@@ -457,10 +458,35 @@ def _get_vhdr_info(vhdr_fname, eog, misc, scale, montage):
         fmt = dict((key, cfg.get('ASCII Infos', key))
                    for key in cfg.options('ASCII Infos'))
 
-    # locate EEG and marker files
+    # locate EEG binary file and marker file for the stim channel
     path = op.dirname(vhdr_fname)
     data_filename = op.join(path, cfg.get('Common Infos', 'DataFile'))
-    info['meas_date'] = int(time.time())
+    mrk_fname = op.join(path, cfg.get('Common Infos', 'MarkerFile'))
+
+    # Try to get measurement date from marker file
+    # Usually saved with a marker "New Segment", see BrainVision documentation
+    regexp = r'^Mk\d*=New Segment,[.*,]\d*.\d*,\d*,(\d{20})$'
+    with open(mrk_fname, 'r') as tmp_mrk_f:
+        lines = tmp_mrk_f.readlines()
+
+    for line in lines:
+        match = re.findall(regexp, line)
+
+        # Always take first measurement date we find
+        if match:
+            date_str = match[0]
+            meas_date = datetime.strptime(date_str, '%Y%m%d%H%M%S%f')
+
+            # We need list of unix time in milliseconds and as second entry
+            # the additional amount of microseconds
+            epoch = datetime.utcfromtimestamp(0)
+            unix_time_millis = (meas_date - epoch).total_seconds()
+            info['meas_date'] = [int(t) for t in
+                                 str(unix_time_millis).split('.')]
+            break
+
+    if not match:
+        info['meas_date'] = DATE_NONE
```

Comment:
you do not need this, just use `else` clause of the `for` loop as:
```
for ...
    ...
else:
    ...
```

Reply:
you do not need the `if not match` I mean

Reply:
wow, a `for-else` loop ... didn't even know that exists. Thanks for pointing it out!